catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/EventProcessorImpl.java

```diff
@@ -170,17 +174,15 @@ public static void processEntry(Metacard metacard, String operation, EventAdmin
 
       // CONTEXTUAL INFORMATION
       if (metacard.getMetadata() != null) {
-        try {
-          // Build Lucene search index on entry's entire metadata using
-          // default XPaths (specified
-          // in ContextualEvaluator) - this index will be used by all
-          // contextual predicates that do
-          // *NOT* specify any textPaths. (Building index here optimizes
-          // code so that this index is
-          // not built for every contextual subscription that has no
-          // textPaths.)
-          Directory index = ContextualEvaluator.buildIndex(metacard.getMetadata());
-
+        // Build Lucene search index on entry's entire metadata using
+        // default XPaths (specified
+        // in ContextualEvaluator) - this index will be used by all
+        // contextual predicates that do
+        // *NOT* specify any textPaths. (Building index here optimizes
+        // code so that this index is
+        // not built for every contextual subscription that has no
+        // textPaths.)
+        try (Directory index = ContextualEvaluator.buildIndex(metacard.getMetadata())) {
```

Comment:
❓ Is it ok to close the index and yet put it in a map that is stored externally? I suspect it ends up being accessed from outside this try()

Reply:
hmm, not sure. I'll ask around

Reply:
Richard and I talked and he basically said that the Directory class will close in future methods because it will use the RAMDirectory class. I'll take it out